packages/ethereum-swap-provider/lib/EthereumSwapProvider.js

```diff
@@ -70,31 +70,48 @@ export default class EthereumSwapProvider extends Provider {
       '60', '48', // PUSH1 48
       'f1', // CALL
 
+      // Validate input size
+      '36', // CALLDATASIZE
+      '60', '20', // PUSH1 20 (32 bytes)
+      '14', // EQ
+      '16', // AND (to make sure SHA256 CALL succeeded)
+
       // Validate with secretHash
       '7f', secretHash, // PUSH32 {secretHashEncoded}
       '60', '21', // PUSH1 21
       '51', // MLOAD
       '14', // EQ
-      '16', // AND (to make sure CALL succeeded)
+      '16', // AND (to make sure input valid size)
       // Redeem if secret is valid
-      '60', '47', // PUSH1 {redeemDestination}
+      '60', '4c', // PUSH1 {redeemDestination}
       '57', // JUMPI
 
       // Check time lock
       '64', expirationEncoded, // PUSH5 {expirationEncoded}
```

Comment:
We check for input size in refund section in ETH contract. Should we do the same here?

Reply:
You mean erc20? 
We can but it might get complicated, refund case doesn't need any guards

Reply:
going to add it
good to be consistent and i think we have the same issue in finding refund